src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java

```diff
@@ -180,10 +178,18 @@ public void filter(FacesContext context, DataTable table, List<FilterMeta> filte
         Locale filterLocale = table.resolveDataLocale();
         boolean hasGlobalFilter = !LangUtils.isValueBlank(globalFilterValue);
         GlobalFilterConstraint globalFilterConstraint = (GlobalFilterConstraint) FILTER_CONSTRAINTS.get(GLOBAL_MODE);
+        MethodExpression globalFilterFunction = table.getGlobalFilterFunction();
         ELContext elContext = context.getELContext();
 
         for (int i = 0; i < table.getRowCount(); i++) {
             table.setRowIndex(i);
+
+            if (globalFilterFunction != null
+                    && (Boolean) globalFilterFunction.invoke(elContext, new Object[]{table.getRowData(), globalFilterValue, filterLocale})) {
+                filteredData.add(table.getRowData());
+                continue;
```

Comment:
This has kept me thinking.. I think it's wrong. Like this you cannot combine the global filter with column filters. Let me revise this.